orca/src/ldap.rs

```diff
@@ -197,26 +222,21 @@ impl LdapClient {
             op: LdapOp::SearchRequest(req),
         };
 
-        let start = Instant::now();
-
-        let count = AtomicUsize::new(0);
-
         // Send it
         let _ = inner.framed.send(msg).await.map_err(|e| {
             error!("Unable to search -> {:?}", e);
         })?;
 
+        let mut results = Vec::new();
         // It takes a lot more work to process a response from ldap :(
         while let Some(Ok(msg)) = inner.framed.next().await {
             match msg.op {
-                LdapOp::SearchResultEntry(_) => {
-                    count.fetch_add(1, Ordering::Relaxed);
-                }
+                LdapOp::SearchResultEntry(ent) => results.push(ent),
```

Comment:
So we don't need Atomic operation for this anymore?

Reply:
Nope, because we return the results now so we can count the length that way instead.